src/test/scala/com/trueaccord/scalapb/json/JsonFormatSpec.scala

```diff
@@ -216,8 +220,29 @@ class JsonFormatSpec extends FlatSpec with MustMatchers with OptionValues {
     new Printer(formattingLongAsNumber = true).print(MyTest(bazinga = Some(642))) must be("""{"bazinga":642}""")
   }
 
-  "TestProto" should "parse a number formatted as JSON string" in {
-    new Parser().fromJsonString[MyTest]("""{"bazinga":642}""") must be(MyTest(bazinga = Some(642)))
+  "TestProto" should "parse numbers formatted as JSON string" in {
+    val parser = new Parser()
+    def validate(json: String, expected: MyTest): Assertion = {
+      parser.fromJsonString[MyTest](json) must be(expected)
+    }
+    def validateException(json: String): Assertion = {
+      a [JsonFormatException] mustBe thrownBy { parser.fromJsonString[MyTest](json) }
+    }
+
+    validate("""{"foobar":"642"}""", MyTest(foobar = Some(642)))
+    validate("""{"foobar":"-1"}""", MyTest(foobar = Some(-1)))
+    validateException("""{"foobar":"10881926652273085354"}""")
+
+    validate("""{"bazinga":"642"}""", MyTest(bazinga = Some(642)))
+    validate("""{"bazinga":"-1"}""", MyTest(bazinga = Some(-1l)))
+    validate("""{"bazinga":"10881926652273085354"}""", MyTest(bazinga = Some(-7564817421436466262l)))
```

Comment:
These two values will throw an exception since they are beyond Long. I believe this happens because you are not using `toLongExact`.